tests/irc/core/test-irc.c

```diff
@@ -0,0 +1,231 @@
+
+typedef struct {
+	char const *const description;
+	char const *const input;
+	char const *const output0;
+	char const *const output1;
+} event_get_params_test_case;
+
+event_get_params_test_case const event_get_params_fixtures[] = {
+	{
+		.description = "Only a channel",
+		.input       = "#test",
+		.output0     = "#test",
+		.output1     = "",
+	},
+	{
+		.description = "Only a channel with trailing space",
+		.input       = "#test ",
+		.output0     = "#test",
+		.output1     = "",
+	},
+	{
+		.description = "No :<trailing>, channel mode with one parameter after channel name",
+		.input       = "#test +i",
+		.output0     = "#test",
+		.output1     = "+i",
+	},
+	{
+		.description = "No :<trailing>, channel mode with two parameters after channel name",
+		.input       = "#test +o tester",
+		.output0     = "#test",
+		.output1     = "+o tester",
+	},
+	{
+		.description = "No :<trailing>, channel mode with three parameters afer channel name",
+		.input       = "#test +ov tester tester2",
+		.output0     = "#test",
+		.output1     = "+ov tester tester2",
+	},
+	{
+		.description = "No :<trailing>, channel mode with three parameters afer channel name, bunch of extra space",
+		.input       = "#test  +ov  tester  tester2 ",
+		.output0     = "#test",
+		.output1     = " +ov  tester  tester2 ",
+	},
+	{
+		.description = "Channel mode with one parameter after channel name, :<trailing> at the start of modes",
+		.input       = "#test :+i",
+		.output0     = "#test",
+		.output1     = "+i",
+	},
+	{
+		.description = "Channel mode with two parameters after channel name, :<trailing> at the  start of modes",
+		.input       = "#test :+o tester",
+		.output0     = "#test",
+		.output1     = "+o tester",
+	},
+	{
+		.description = "Channel mode with three parameters after channel name, :<trailing> at the start of modes",
+		.input       = "#test :+ov tester tester2",
+		.output0     = "#test",
+		.output1     = "+ov tester tester2",
+	},
+	{
+		.description = "Channel mode with two parameters after channel name, :<trailing> on the final parameter",
+		.input       = "#test +o :tester",
+		.output0     = "#test",
+		.output1     = "+o tester",
+	},
+	{
+		.description = "Channel mode with three parameters after channel name, :<trailing> on the final parameter",
+		.input       = "#test +ov tester :tester2",
+		.output0     = "#test",
+		.output1     = "+ov tester tester2",
+	},
+	{
+		.description = "Channel mode with three parameters after channel name, :<trailing> on the final parameter, also a second : present",
+		.input       = "#test +ov tester :tester2 hi:there",
+		.output0     = "#test",
+		.output1     = "+ov tester tester2 hi:there",
+	},
+};
+
+static void test_event_get_params(const event_get_params_test_case *test);
+
+int main(int argc, char **argv)
+{
+	int i;
+
+	g_test_init(&argc, &argv, NULL);
+
+	for (i = 0; i < G_N_ELEMENTS(event_get_param_fixtures); i++) {
```

Comment:
I think we want to iterate over the other array here. That's why some of the test cases aren't running.

Reply:
omg! good catch